overlord/snapstate/handlers.go

```diff
@@ -1363,12 +1363,13 @@ func (m *SnapManager) doUnlinkCurrentSnap(t *state.Task, _ *tomb.Tomb) (err erro
 	// symlink to a new revision of the snapd snap, so only do the actual
 	// unlink if we're not working on the snapd snap
 	if oldInfo.Type() != snap.TypeSnapd {
+		// establish a persistent run inhibition lock
+		if err := m.backend.InhibitSnap(oldInfo.InstanceName(), oldInfo.Revision, runinhibit.HintInhibitedForRefresh); err != nil {
+			return err
+		}
```

Comment:
I think it doesn't add any new failure modes but the new behaviour in the error case isn't the same as before. UnlinkSnap would always try to unlink even if the inhibit failed. However, the worst that can happen is that the snap stays inhibited but not unlinked and that could already happen before (I think)

Reply:
Great point, I don't think that this new mode would cause issue because now `LinkSnap`/`UnlinkSnap` are using ensure behavior so if `InhibitSnap`, the undo operation will call `LinkSnap` which would be effectively a no-op since `UnlinkSnap` was not called in the do direction.

Anyway, I totally agree that it would be great to change as little as possible, I will try to move it to the old behavior.